src/plots/cartesian/transition_axes.js

```diff
@@ -100,7 +100,20 @@ module.exports = function transitionAxes(gd, newLayout, transitionOpts, makeOnCo
     var updatedAxisIds = Object.keys(updates);
     var affectedSubplots = computeAffectedSubplots(fullLayout, updatedAxisIds, updates);
 
+    function updateLayoutObjs() {
+        function redrawObjs(objArray, method) {
+            for(var i = 0; i < objArray.length; i++) {
+                method(gd, i);
+            }
+        }
+
+        redrawObjs(fullLayout.annotations || [], Registry.getComponentMethod('annotations', 'drawOne'));
+        redrawObjs(fullLayout.shapes || [], Registry.getComponentMethod('shapes', 'drawOne'));
+        redrawObjs(fullLayout.images || [], Registry.getComponentMethod('images', 'draw'));
```

Comment:
Looking at [`dragbox.js`](https://github.com/plotly/plotly.js/blob/9948af4f345fac61280677f367d67a4599115c4a/src/plots/cartesian/dragbox.js#L580), images can short circuit that loop over `objArray` above.